cmd/cmd.go

```diff
@@ -67,6 +94,56 @@ var (
 	gaugeVersion    bool
 )
 
+func handleRepeatCommand(cmd *cobra.Command) {
+	if repeat {
+		prevCmd := readPrevCmd()
+		executeCmd(cmd, prevCmd.Command)
+	} else {
+		if prevFailed {
+			prevFailed = false
+			return
+		}
+		writePrevCmd()
+	}
+}
+
+func executeCmd(cmd *cobra.Command, lastState []string) {
+	cmd.Parent().SetArgs(lastState[1:])
+	os.Args = lastState
+	resetFlags()
+	cmd.Execute()
+}
+
+func readPrevCmd() *prevCommand {
+	contents, err := common.ReadFileContents(filepath.Join(config.ProjectRoot, rerun.DotGauge, lastRunCmdFileName))
+	if err != nil {
+		logger.Fatalf("Failed to read previous command information. Reason: %s", err.Error())
+	}
+	meta := newPrevCommand()
+	if err = json.Unmarshal([]byte(contents), meta); err != nil {
+		logger.Fatalf("Invalid previous command information. Reason: %s", err.Error())
+	}
+	return meta
+}
+
+func writePrevCmd() {
+	prevCmd := newPrevCommand()
+	prevCmd.Command = os.Args
+	contents, err := prevCmd.getJSON()
+	if err != nil {
+		logger.Fatalf("Unable to parse last run command. Error : %v", err.Error())
+	}
+	prevCmdFile := filepath.Join(config.ProjectRoot, rerun.DotGauge, lastRunCmdFileName)
+	dotGaugeDir := filepath.Join(config.ProjectRoot, rerun.DotGauge)
```

Comment:
There is already a constant for .gauge folder defined in common. We should be using that variable everywhere.